lib/encoder/src/Pololu37D.cpp

```diff
@@ -5,23 +5,55 @@
 using namespace Encoder;
 
 Pololu37D::Pololu37D(const Config &config)
-    : m_QEI(config.ChannelA, config.ChannelB, config.Index, config.quadratureEncoding),
+    : m_QEI(config.ChannelA, config.ChannelB, config.Index, GPIO::QEI::Encoding::X4_ENCODING),
       m_zeroOffsetDeg(config.offsetDeg) {}
 
 bool Pololu37D::getAngleDeg(float &angle) {
   std::scoped_lock<Mutex> lock(m_mutex);
-  angle = (m_QEI.getPulses() * m_degreesPerCount) - m_zeroOffsetDeg;
+  if (read()) {
+    // reading was succesful
+    angle = m_current_angle_deg;
+  } else {
+    return false;
+  }
   return true;
 }
 
 bool Pololu37D::getAngularVelocityDegPerSec(float &speed) {
   std::scoped_lock<Mutex> lock(m_mutex);
-  speed = m_QEI.getPulseVelocity_PulsesPerSec() * m_degreesPerCount;
+  read();
+  speed = m_anglular_velocity_deg_per_sec;
   return true;
 }
 
 bool Pololu37D::reset() {
   std::scoped_lock<Mutex> lock(m_mutex);
+  m_zeroOffsetDeg                 = 0.0;
+  m_current_angle_deg             = 0.0;
+  m_previous_angle_deg            = 0.0;
+  m_anglular_velocity_deg_per_sec = 0.0;
   m_QEI.reset();
   return true;
 }
+
+bool Pololu37D::read() {
```

Comment:
if this internal function always succeeds just make it void read()

Reply:
done